Source/DafnyCore/Generic/ReporterExtensions.cs

```diff
@@ -55,14 +55,18 @@ private static string FormatRelated(string related) {
   public static IEnumerable<DafnyRelatedInformation> CreateDiagnosticRelatedInformationFor(IOrigin token, string? message, bool usingSnippets) {
     var (tokenForMessage, inner, newMessage) = token is NestedOrigin nestedToken ? (nestedToken.Outer, nestedToken.Inner, nestedToken.Message) : (token, null, null);
     var dafnyToken = BoogieGenerator.ToDafnyToken(true, tokenForMessage);
-    if (!usingSnippets && dafnyToken.IncludesRange) {
-      if (message == PostConditionFailingMessage) {
-        var postcondition = dafnyToken.PrintOriginal();
-        message = $"this postcondition might not hold: {postcondition}";
-      } else if (message == null || message == RelatedLocationMessage) {
-        message = FormatRelated(dafnyToken.PrintOriginal());
-      }
-    }
+
+    // Turning this on changes many regression tests, in a way that might be considered good,
+    // but it should be turned on in a separate PR
+    // There seem to be no LSP tests for this behavior, so turning it off did not affect those.
```

Comment:
Can you please elaborate on commenting out this?
Normally, if something is not tested, one would add tests, mea culpa if I did not do this when I initially wrote that part.
But I would like to understand why you decided to comment it out. It renders the post condition directly into the message when not using snippets, which is what the LSP is doing.
Did you verify that failing postconditions are still displayed when using the LSP? Perhaps it's redundant and that's why you could remove it.

Reply:
Since it's not tested, it's not clear to me when this currently triggers in the IDE, so if I comment it back in I'm not sure what I'm changing on the IDE side, and I'll be changing a lot on the CLI side. If I do not want to change the CLI then I need to add something specific to detect that the IDE and not the CLI is running, and I have intentionally not made any such information available in `DafnyOptions` to force the two to be more consistent.

Would you be OK with a follow-up PR to comment this back in, add an IDE test and update the CLI tests?